src/engine/QueryPlanner.cpp

```diff
@@ -834,15 +832,14 @@ vector<QueryPlanner::SubtreePlan> QueryPlanner::getDistinctRow(
   const vector<SubtreePlan>& previous = dpTab[dpTab.size() - 1];
   vector<SubtreePlan> added;
   added.reserve(previous.size());
-  for (size_t i = 0; i < previous.size(); ++i) {
-    const SubtreePlan& parent = previous[i];
+  for (const auto& i : previous) {
```

Comment:
I like the range-based for loop,
but I think it should be `const auto& parent`, not `i` (as in the previous code)